### ISSUE TITLE
Add Anthropic Inference Client

### ISSUE THREAD

**Issue by sutt** [sutt/lime#1]
Currently, the tool can call LocalInference, or OpenAI. 

Add the capabilities for the tool to call Anthropic / Claude models:
- [ ] Pass this into all necessary modules, e.g. Model Interface. Let's use `claude` as the alias prefix for selecting it on command line.
https://github.com/sutt/lime/blob/0137ad0c27392f579d0f450ab1c41bdded94e99d/lime/common/inference/interface.py#L25-L42
- [ ] Add any neccesary dependencies to `setup.py` and other config files.
- [ ] Add a way to collect the Anthropic api-key from config files, e,g 
https://github.com/sutt/lime/blob/0137ad0c27392f579d0f450ab1c41bdded94e99d/lime/common/models/state.py#L30-L35
- [ ] Pass the existing inference params into the client's request (mapping the param names if necessary)
https://github.com/sutt/lime/blob/0137ad0c27392f579d0f450ab1c41bdded94e99d/lime/common/models/config.py#L88-L92
- [ ] Build at least one test that has a mock response from Anthropic
https://github.com/sutt/lime/blob/0137ad0c27392f579d0f450ab1c41bdded94e99d/tests/test_eval.py#L40

**In summary:** this PR should enable us to run the commands like `lime eval -m claude-3.5 .` and have it do inference off the calude api using the correct inference params

**Comment by aliraza556** [comment]
Hi @sutt, Please review this PR #2

**Comment by sutt** [comment]
Yes
